### PR TITLE
adds illustrations to help understand attribute service and building steps

### PR DESCRIPTION
**Pull request by JBAhire**
This PR
- Answers how Hypertrace uses attribute service
- Adds schematic representation for attribute service 
- Adds flows involving attribute service
- Add steps to build image locally
- link to dockerhub image source

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# attribute-service
+# Attribute-service
```
Comment (avinashkolluru) [inline README.md:1]: nit: upper case A followed by a `-service` doesn't feel right

File: README.md [hunk 2]
```diff
@@ -12,3 +12,31 @@ for the structure of an attribute.
 ### How are attributes created?
 An initial list of attributes needed by Hypertrace are seeded from `helm/configs` but they
 can also be dynamically registered and queried using the APIs of AttributeService.
+
+## How Hypertrace uses Attribute service?
+Below is the high level schematic diagram(of the push based model) of different components involved in attribute service:
+
+| ![space-1.jpg](https://imagizer.imageshack.com/v2/xq90/924/xN2Top.png) |
```
Comment (avinashkolluru) [inline README.md:19]: These images are slightly outdated. Needs to be fixed before we add them here.

**Review by avinashkolluru — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (JBAhire) [reply README.md:19]: ohh! Who can you help me to get those updated @avinashkolluru ?

File: README.md (hunk 1 above)
Comment (JBAhire) [reply README.md:1]: Okay! will make it `Attribute service`

File: README.md (hunk 2 above)
Comment (JBAhire) [reply README.md:19]: I will remove those for now and we can update them and then add in README.

**Comment by avinashkolluru** [comment]
@adriancole @buchi-busireddy rleiwang is still a code owner. 
We should clean that up first.

**Review by adriancole — COMMENTED** [review]
* What are Attributes?
  * please add a concrete example and switch the text format of the code link to an actual link.
* Add section primary use cases
  * There is mostly what not why here.. I can guess this allows things like auto-complete, what else works or doesn't work based on this (in the UI as there's no anomaly detection in OSS)
* High level diagram
  * use the same name. ex "attribute-service" will match the README "Global Attribute Service" will not
  * use real names for "domain-entity-gateway" and "Span/Trace Enrichment" as no one will know what these are as they are not services. Also I think auto-complete is likely routed through graphql, not calling directly anything. Make the diagram accurate in this regard until it is not an issue.
  * In text before the image you can make sentence saying "Here is the use of attribute-service in context of its callers: [hypertrace-graphql](link to github), [entity-service](link to github) and [hypertrace-trace-enricher](link to github)" Even if consolidation happens later, this is more concrete.
  * Write path is not precise, show exactly what is happening.. there are write apis used only by certain services visible in OSS.

service schematic representation:
 * same feedback as above use real container names.
 * remove anything not in OSS such as anomaly service, and make sure the flow is actually valid

File: README.md (hunk 1 above)
Comment (adriancole) [reply README.md]: make it match deployments "attribute-service" because this is literally something people will see in logs etc

**Comment by adriancole** [comment]
> @adriancole @buchi-busireddy rleiwang is still a code owner.
> We should clean that up first.

that's independent I think and can go here or another PR or commit to master doesnt' matter. If any of this happens @JBAhire can rebase if that helps or just ignore for this PR

File: README.md [hunk 3]
```diff
@@ -1,14 +1,50 @@
...
-Service to provide CRUD operations for the attributes in Hypertrace.
+Attribute service provides CRUD operations for the attributes in Hypertrace.
 
-### What are Attributes?
-An attribute represents a piece of data that's present in Hypertrace
-that's available for querying. Attribute metadata gives the type of the
-data, kind of aggregations allowed on it, services which serve that attribute, etc.
-See `attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto`
-for the structure of an attribute.
+It fetches all attributes relevant to the scope of what is being shown on UI. The concept of Attributes helps build a generic framework of communication between the UI and backend.
```
Comment (avinashkolluru) [inline README.md:5]: Line 5 should be removed

File: README.md [hunk 4]
```diff
@@ -1,14 +1,50 @@
...
+
+## What are Attributes?
+An attribute represents a piece of data that's present in Hypertrace that's available for querying. It is a highly generic building block of what queries/show/display/represent in the UI. Attribute metadata gives the type of the data, kind of aggregations allowed on it, services which serve that attribute, etc.
```
Comment (avinashkolluru) [inline README.md:8]:
```suggestion
An attribute is a granular piece of information that is present in Hypertrace and available for querying/displaying in the UI. An attribute encapsulates information like 1) the type of the data 2) kind of aggregations allowed on it, 3) underlying service/source that can serve this data etc.
This helps in building a generic framework of communication between the UI and the backend and also assists the backend to fetch data from the appropriate sources in a generic way.
```

**Review by avinashkolluru — DISMISSED** [review]

**Comment by kotharironak** [comment]
@JBAhire To fix snyk issue, update below library version from 0.1.3 to 0.2.0 in below files
From:
```
./attribute-service-impl/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.1.3")
./attribute-service/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.1.3")
./attribute-service-client/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.1.3")
```
To:
```
./attribute-service-impl/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.2.0")
./attribute-service/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.2.0")
./attribute-service-client/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.2.0")
```

**Comment by JBAhire** [comment]
> @JBAhire To fix snyk issue, update below library version from 0.1.3 to 0.2.0 in below files
> From:
> 
> ```
> ./attribute-service-impl/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.1.3")
> ./attribute-service/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.1.3")
> ./attribute-service-client/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.1.3")
> ```
> 
> To:
> 
> ```
> ./attribute-service-impl/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.2.0")
> ./attribute-service/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.2.0")
> ./attribute-service-client/build.gradle.kts:  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.2.0")
> ```

Thanks, Ronak! Let me try that!

**Review by avinashkolluru — APPROVED** [review]

**Pull request merged**